ERCS/erc-7683.md

```diff
@@ -98,79 +118,241 @@ struct Input {
+
+Let there be a sub-type called `Message`, which is defined by the following structs:
+
+```solidity
+// The Message subtype allows ERC7683 intents to carry calldata that is executed on a target contract on the destination chain. The settlement contract that the filler interacts with on the destination chain will decode the message into smart contract calls and execute the calls within the filler's `fill()` transaction.
+
+// The Message contains calls that the user wants executed on the destination chain.
+// The target is a contract on the destination chain that the settlement contract will attempt to send callData and value to.
+struct Calls {
+  address target;
+  bytes callData;
+  uint256 value;
+}
+
+struct Message {
+  Calls[] calls;
+}
+```
+
+The `Message` sub-type is designed to be used by a 7683 user to incentivize a filler to to execute arbitrary calldata on a target destination chain contracton the user's behalf. For example, the settlement contract might decode the `orderData` containing the message information as follows:
+
+```solidity
+function fill(bytes32 orderId, bytes calldata originData, bytes calldata fillerData) public {
+	(
+		address user,
+		uint32 fillDeadline,
+		Output memory fillerOutput,
+		Message memory message
+	) = abi.decode(originData);
+
+	// ...Do some preprocessing on the parameters here to validate the order...
+
+	// ...Execute the fill logic of the ResolvedCrossChainOrder...
+
+	// Handle the Message subtype:
+
+        // Revert if any of the message calls fail.
+        uint256 length = message.calls.length;
+        for (uint256 i = 0; i < length; ++i) {
+            Call memory call = message.calls[i];
+
+            // If we are calling an EOA with calldata, assume target was incorrectly specified and revert.
+            if (call.callData.length > 0 && call.target.code.length == 0) {
+                revert InvalidCall(i, calls);
+            }
+
+            (bool success, ) = call.target.call{ value: call.value }(call.callData);
+            if (!success) revert CallReverted(i, message.calls);
+        }
+    }
+```
+
+In this example, the Message sub-type allows the user to delegate destination chain contract execution to fillers. However, because transactions are executed via filler, the `msg.sender` would be equal to the filler's account, making this `Message` sub-type limited if the target contract authenticates based on the `msg.sender`. Ideally, 7683 orders containing Messages can be combined with smart contract wallets like implementations of [ERC4337](https://github.com/across-protocol/ERCs/blob/master/ERCS/erc-4337.md) or [EIP7702](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md) to allow complete cross-chain delegated execution.
```

Comment:
wouldnt the sender here be the `IDestinationSettler` rather than the filler's account?

Reply:
You're absolutely right. This is a mistake.

Reply:
Fixed.